doc/streams-dsl.md

```diff
@@ -0,0 +1,78 @@
+# Kafka Streams DSL
+
+## Rationale
+
+The Jackdaw Streams API is a thin wrapper around the underlying Kafka Streams
+DSL. It allows the definition of kafka streams applications using idiomatic
```

Comment:
The official branding is "Kafka Streams", the "Kafka Streams API", and sometimes "Streams API" (but not, "DSL"). If we want to draw a distinction between that and Jackdaw, we could call it the "Java Kafka Streams API".

Reply:
What documentation do you mean? It is referred to as a DSL in title of the streams sections in both kafka and confluent documentation.

https://docs.confluent.io/current/streams/developer-guide/dsl-api.html
https://kafka.apache.org/21/documentation/streams/developer-guide/dsl-api.html